lib/net/dhcp/record.rb

```diff
@@ -57,8 +70,10 @@ def ==(other)
 
       # If we're converting an 'ad-hoc' lease created by a host booting outside of Foreman's knowledge,
       # then :hostname will be blank on the incoming lease - if the ip/mac still match, then this
-      # isn't a conflict.
-      to_compare << :hostname if other.attrs[:hostname].present? && attrs[:hostname].present?
+      # isn't a conflict. Only applicable on legacy proxy API without "type" attribute.
+      if legacy_dhcp_api?
+        to_compare << :hostname if other.attrs[:hostname].present? && attrs[:hostname].present?
+      end
```

Comment:
Moved the old hostname-based comparison to legacy branch. Hostname is no longer relevant, it's just a "comment" type of entry in DHCP.